File: app/src/main/java/com/bytedance/tiktok/adapter/PrivateLetterAdapter.java
package com.bytedance.tiktok.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import com.bytedance.tiktok.R;
import com.bytedance.tiktok.base.BaseRvAdapter;
import com.bytedance.tiktok.base.BaseRvViewHolder;
import com.bytedance.tiktok.bean.VideoBean;

import java.util.List;

import butterknife.BindView;

/**
 * create by libo
 * create on 2020-05-25
 * description
 */
public class PrivateLetterAdapter extends BaseRvAdapter<VideoBean.UserBean, PrivateLetterAdapter.PrivateLetterViewHolder> {

    public PrivateLetterAdapter(Context context, List<VideoBean.UserBean> datas) {
        super(context, datas);
    }

    @Override
    protected void onBindData(PrivateLetterViewHolder holder, VideoBean.UserBean userBean, int position) {
        holder.ivHead.setImageResource(userBean.getHead());
        holder.tvName.setText(userBean.getNickName());
    }

    @NonNull
    @Override
    public PrivateLetterViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_private_letter,parent, false);
        return new PrivateLetterViewHolder(view);
    }

    public class PrivateLetterViewHolder extends BaseRvViewHolder {
        @BindView(R.id.iv_head)
        ImageView ivHead;
        @BindView(R.id.tv_nickname)
        TextView tvName;

        public PrivateLetterViewHolder(View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/com/bytedance/tiktok/adapter/PrivateLetterAdapter.kt
package com.bytedance.tiktok.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bytedance.tiktok.adapter.PrivateLetterAdapter.PrivateLetterViewHolder
import com.bytedance.tiktok.base.BaseAdapter
import com.bytedance.tiktok.bean.VideoBean.UserBean
import com.bytedance.tiktok.databinding.ItemPrivateLetterBinding

/**
 * create by libo
 * create on 2020-05-25
 * description
 */
class PrivateLetterAdapter : BaseAdapter<PrivateLetterViewHolder, UserBean>(FansDiff()) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PrivateLetterViewHolder {
        return PrivateLetterViewHolder(ItemPrivateLetterBinding.inflate(LayoutInflater.from(parent.context), parent, false))
    }

    override fun onBindViewHolder(holder: PrivateLetterViewHolder, position: Int) {
        var userBean = mList[position]
        holder?.binding?.ivHead!!.setImageResource(userBean!!.head)
        holder?.binding.tvNickname!!.text = userBean?.nickName
    }

    inner class PrivateLetterViewHolder(val binding: ItemPrivateLetterBinding) : RecyclerView.ViewHolder(binding.root) {

    }
}